### PR TITLE
Add dispatch for `cudf.Dataframe` to/from `pyarrow.Table` conversion

### PR DESCRIPTION
**Pull request by rjzamora**
## Description
Registers the necessary cudf-specific logic for the dispatch functions introduced in https://github.com/dask/dask/pull/10312. These dispatch functions are necessary to enable "p2p" shuffling with a cudf backend (see: https://github.com/dask/distributed/pull/7743)

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: python/dask_cudf/dask_cudf/backends.py [hunk 1]
```diff
@@ -372,6 +372,36 @@ def percentile_cudf(a, q, interpolation="linear"):
...
+            warnings.warn(
+                f"Ignoring the following arguments to "
+                f"`to_pyarrow_table_dispatch`: {list(kwargs)}"
+            )
+        return obj.to_arrow(preserve_index=preserve_index)
+
+    @from_pyarrow_table_dispatch.register(cudf.DataFrame)
+    def _table_to_cudf(obj, table, **kwargs):
+        # cudf must ignore self_destruct
+        kwargs.pop("self_destruct", None)
```
Comment (wence-) [inline python/dask_cudf/dask_cudf/backends.py:394]:
```suggestion
    def _table_to_cudf(obj, table, self_destruct=None, **kwargs):
        # cudf ignores self_destruct.
```
?

**Review by wence- — APPROVED** [review]
Minor, but looks good, thanks @rjzamora

File: python/dask_cudf/dask_cudf/backends.py (hunk 1 above)
Comment (rjzamora) [reply python/dask_cudf/dask_cudf/backends.py]: works for me

**Pull request merged**